receiver/prometheusremotewritereceiver/receiver.go

```diff
@@ -191,17 +193,75 @@ func (prw *prometheusRemoteWriteReceiver) translateV2(_ context.Context, req *wr
 			intraRequestCache[hashedLabels] = rm
 		}
 
+		scopeName, scopeVersion := prw.extractScopeInfo(ls)
+		metricName := ls.Get(labels.MetricName)
+		unit := ""
+		// TODO: Like UnitRef, we should assign the HelpRef to the metric.
+		if ts.Metadata.UnitRef > 0 && ts.Metadata.UnitRef < uint32(len(req.Symbols)) {
+			unit = req.Symbols[ts.Metadata.UnitRef]
+		}
+
+		// Temporary approach to generate the metric key.
+		// TODO: Replace this with a proper hashing function.
+		metricKey := fmt.Sprintf("%d:%s:%s:%s:%s:%d",
+			hashedLabels,     // Resource identity
```

Comment:
instead of use hashedLabels, use the identity library over the resource

Reply:
Documentation reference: https://opentelemetry.io/docs/specs/otel/metrics/data-model/#opentelemetry-protocol-data-model

Reply:
https://opentelemetry.io/docs/specs/otel/metrics/data-model/#opentelemetry-protocol-data-model:~:text=The%20data%20point%20type%2C%20unit%2C%20and%20intrinsic%20properties%20are%20considered%20identifying%2C%20whereas%20the%20description%20field%20is%20explicitly%20not%20identifying%20in%20nature